android/Dockerfile

```diff
@@ -22,20 +32,25 @@ RUN dnf install -y \
     && rm -rf /var/cache/dnf
 
 FROM tools as android
-ARG ANDROID_TOOLS=8092744
-RUN mkdir -p /opt/android && cd /opt/android && wget "https://dl.google.com/android/repository/commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" \
-    && unzip "commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" && rm "commandlinetools-linux-${ANDROID_TOOLS}_latest.zip"
-ENV JAVA_HOME /usr/lib/jvm/java
-RUN yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux --update \
-    && yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "platform-tools" "tools"
-ARG ANDROID_LEVEL=31
-RUN yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "platforms;android-${ANDROID_LEVEL}"
-ARG ANDROID_BUILD_TOOLS=31.0.0
-RUN yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "build-tools;${ANDROID_BUILD_TOOLS}"
-ARG NDK=23.1.7779620
-RUN yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux --update \
-    && yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "ndk;${NDK}"
-RUN find /opt/android-sdk-linux/ndk -name libz.a -delete
+ARG ANDROID_TOOLS
+ARG ANDROID_LEVEL
+ARG ANDROID_BUILD_TOOLS
+ARG NDK
+ARG JAVA_HOME_ARG
+ENV JAVA_HOME=$JAVA_HOME_ARG
+RUN mkdir -p /opt/android && \
+    cd /opt/android && \
+    wget "https://dl.google.com/android/repository/commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" && \
+    unzip "commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" && \
+    rm "commandlinetools-linux-${ANDROID_TOOLS}_latest.zip" && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux --update && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "ndk;${NDK}" && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "platforms;android-${ANDROID_LEVEL}" && \
+    yes | /opt/android/cmdline-tools/bin/sdkmanager --sdk_root=/opt/android-sdk-linux "build-tools;${ANDROID_BUILD_TOOLS}" && \
+    find /opt/android-sdk-linux/ndk -name libz.a -delete
+#optimize image, the rest is not needed for now
```

Comment:
Can these lines be removed entirely if they are not needed?

Reply:
I left those as reference for the time we start working on running android tests on qemu.
Please treat this PR as an iterative improvement.